.github/workflows/scanner-offline-bundle-update.yaml

```diff
@@ -40,31 +51,74 @@ jobs:
       with:
         gcp-account: ${{ secrets.GCP_SERVICE_ACCOUNT_STACKROX_CI }}
 
-    - name: Authenticate with Google Cloud
+    - name: Generate offline bundle
+      run: |
+        .github/workflows/scripts/scanner-update-offline-bundle.sh \
+            bundle \
+            "$VULNERABILITY_VERSION" \
+            "$SUPPORTED_RELEASES"
+
+    - uses: ./.github/actions/upload-artifact-with-retry
+      with:
+        name: offline-bundle-${{ matrix.versions.vulnerability_version }}
+        path: bundle
+        if-no-files-found: error
+
+  upload-bundle:
+    runs-on: ubuntu-latest
+    needs:
+    - generate-bundle
+    steps:
+    - uses: actions/download-artifact@v4
+      with:
+        path: bundles
+        pattern: offline-bundle-*
+        merge-multiple: true
+
+    - name: Authenticate with Google Cloud (release)
+      if: github.event_name == 'schedule'
       uses: google-github-actions/auth@v2
       with:
         credentials_json: ${{ secrets.GOOGLE_SA_STACKROX_HUB_VULN_DUMP_UPLOADER }}
 
+    - name: Authenticate with Google Cloud (test)
+      if: github.event_name != 'schedule'
+      uses: google-github-actions/auth@v2
+      with:
+        credentials_json: ${{ secrets.GOOGLE_SA_CIRCLECI_SCANNER }}
+
     - name: Set up Cloud SDK
       uses: google-github-actions/setup-gcloud@v2
 
-    - name: Generate offline vulnerability bundle
+    - name: Upload bundles (schedule)
+      if: github.event_name == 'schedule'
+      run: |
+        cd bundles
+        gsutil cp scanner-v4-defs-*.zip "gs://definitions.stackrox.io/v4/offline-bundles/"
+
+    - name: Upload bundles (pull request)
+      if: github.event_name != 'schedule'
       run: |
-        case "${{ env.ROX_PRODUCT_VERSION }}" in
-        4.4)
-            .github/workflows/scripts/scanner-update-offline-bundle.sh "${{ env.ROX_PRODUCT_VERSION }}" "vulns.json.zst"
-            ;;
-        *)
-            .github/workflows/scripts/scanner-update-offline-bundle.sh "${{ env.ROX_PRODUCT_VERSION }}" "vulnerabilities.zip"
-            ;;
+        branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}
+        # Replace / with -, so the branch name isn't truncated when pushed to GCS.
+        dir=${branch////-}
+        case $dir in
+        dev|1.0.0)
+          echo "Branch $dir is disallowed"
+          exit 1
         esac
 
+        cd bundles
+        ls -lh
+        gsutil cp -r . "gs://scanner-v4-test/offline-bundles/$dir"
+
   send-notification:
     needs:
-    - read-release-versions
-    - upload-release-vulnerabilities
+    - generate-version-matrix
+    - generate-bundle
+    - upload-bundle
     runs-on: ubuntu-latest
-    if: failure()
+    if: github.event_name == 'schedule' && failure()
```

Comment:
same idea. I do also wonder if the other workflows are missing similar checks

Reply:
:+1: